### PR TITLE
feat: add memory extract telemetry breakdown

### PR DESCRIPTION
**Pull request by zhoujh01**
## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Test update

## Changes Made

  核心变更分 4 块：

  - Telemetry 能力增强
      - 给 memory extract 增加了更细的 telemetry breakdown
      - 给 resource 处理链路增加了 telemetry breakdown
      - telemetry summary 会省略值为 0 的字段，返回更干净
  - API / Console 行为更新
      - resources/temp_upload 现在支持返回 telemetry
      - console 上传资源时会把 upload 和 add_resource 两段 telemetry 一起展示
      - 相关后端入口在 openviking/server/routers/resources.py、前端在 openviking/console/static/
        app.js
  - 核心实现调整
      - 改了 openviking/telemetry/operation.py
      - 补充了 openviking/session/memory_extractor.py、openviking/service/resource_service.py、
        openviking/utils/resource_processor.py 的 telemetry 采集
      - 还处理了 openviking/session/compressor.py 的代码整理/冲突结果
      - 新增了用户向的双语 guide：
          - docs/en/guides/07-operation-telemetry.md
          - docs/zh/guides/07-operation-telemetry.md
      - 同步更新了 CONTRIBUTING.md、CONTRIBUTING_CN.md、CONTRIBUTING_JA.md
      - 新增/更新了 telemetry、resources、sessions 相关测试

## Testing

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have tested this on the following platforms:
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows

## Checklist

- [ ] My code follows the project's coding style
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

File: openviking/session/compressor.py [hunk 1]
```diff
@@ -234,208 +235,254 @@ async def extract_long_term_memories(
             return []
 
         self._pending_semantic_changes.clear()
+        telemetry = get_current_telemetry()
+        telemetry.set("memory.extract.candidates.total", 0)
```
Comment (qin-ptr) [inline openviking/session/compressor.py:293]: **[Design]** (blocking)

Telemetry initialization strategy is inconsistent with documentation semantics. Lines 239-245 unconditionally initialize `memory.extract.*` gauges to 0 before calling `extract()`. If `extract()` returns early due to:

1. VLM unavailability (memory_extractor.py:239-241 returns empty list)
2. Empty candidates (line 256 returns early)

The telemetry will still contain `memory.extract` field with all-zero values plus a small duration_ms. This contradicts docs/design/operation_telemetry.md:298: "`summary.memory.extract` 仅在 memory extract 细分指标存在时出现". All-zero fields mislead callers into thinking extract executed but produced nothing, when VLM unavailability may have prevented any attempt.

**Suggested fix**: Check if candidates is empty before initializing telemetry gauges (move lines 239-245 after line 256 check), or add logic in TelemetrySummaryBuilder.build() to omit `memory.extract` if all key metrics are zero, or update documentation to explicitly state all-zero fields may be returned when extract is skipped.

File: openviking/console/static/app.js [hunk 2]
```diff
@@ -1737,6 +1737,7 @@ function bindAddResource() {
           upload: uploadPayload.result,
           add_resource: addPayload.result,
         },
+        telemetry: addPayload.telemetry,
```
Comment (qin-ptr) [inline openviking/console/static/app.js]: **[Bug]** (blocking)

Directly assigning `telemetry: addPayload.telemetry` creates a property with undefined value when telemetry is not requested. While JSON.stringify ignores undefined, for code clarity use conditional inclusion:

```javascript
setOutput({
  status: "ok",
  result: {
    upload: uploadPayload.result,
    add_resource: addPayload.result,
  },
  ...(addPayload.telemetry && { telemetry: addPayload.telemetry }),
});
```

File: tests/server/test_api_sessions.py [hunk 3]
```diff
@@ -167,10 +168,65 @@ async def test_compress_session_with_telemetry(client: httpx.AsyncClient):
     assert summary["operation"] == "session.commit"
     assert {"total", "llm", "embedding"}.issubset(summary["tokens"].keys())
     assert summary["memory"]["extracted"] is not None
+    assert "extract" in summary["memory"]
```
Comment (qin-ptr) [inline tests/server/test_api_sessions.py:171]: **[Suggestion]** (non-blocking)

This assertion assumes `memory.extract` field always exists when session.commit is called with telemetry. However, per documentation semantics (docs line 298), this field should only appear when extract breakdown metrics actually exist.

Consider adding explicit test cases for edge scenarios:
1. Mock VLM unavailable → verify `memory.extract` field behavior (present with zeros? or absent?)
2. Empty candidates list → verify field behavior

This ensures implementation-documentation alignment and prevents regressions.

File: docs/design/operation_telemetry.md [hunk 4]
```diff
@@ -281,19 +437,210 @@ OpenViking 需要一套统一的 telemetry 机制，用来描述一次操作在
...
+- 类型：`integer`
+- 含义：最终被跳过的 candidate 数量
+- 单位：个
+
+#### `summary.memory.extract.stages.prepare_inputs_ms`
+
+- 类型：`number`
+- 含义：LLM 提取前的数据准备耗时
+- 单位：毫秒
+- 范围：ToolPart 收集、消息格式化、语言检测、prompt 渲染
```
Comment (qin-ptr) [inline docs/design/operation_telemetry.md]: **[Suggestion]** (non-blocking)

Format inconsistency: this field uses "范围:" (Range) label while other fields use "备注:" (Notes). For consistency, consider unifying to "备注:" or applying "范围:" to all similar fields describing coverage scope.
